### PR TITLE
fix: change duration of snackbar from short to medium

### LINKED ISSUE #3838: [SnackBar] Allow to change default duration
## Feature description

Currently the default duration of SnackBar is `DURATION.short`. I'd like to change it to `DURATION.medium` globally.

### PR DESCRIPTION
**Pull request by encetroc**
Fixes #3838 

#### Description

- I changed the default duration of snackbar from short to meduim
- I updated the documentation to reflect the changes made

#### Scope

Patch: Bug Fix

### CODE REVIEW

**Comment by chasestarr** [comment]
Thanks for looking into this. Based on our design guidelines snackbars with one line of text should use the short duration. I assume that will be the most common case so the default should remain as short. When working on a particular application I may know that my snackbars more often have two lines. As a developer experience improvement I should be able to supply the `SnackbarProvider` with a `defaultDuration` prop so that I can save some typing and only configure the outliers.

```
<SnackbarProvider />

// this queued snackbar should render for the short duration
enqueue({message: 'hello world'});
```

```
<SnackbarProvider defaultDuration={DURATION.medium} />

// this queued snackbar should render for the medium duration
enqueue({message: 'hello world'});
```

**Comment by encetroc** [comment]
Thanks for your clarification as this is my first time trying to contribute to open source. I updated my PR and added the possibility to change the default duration using the provider.

File: src/snackbar/snackbar-context.js [hunk 1]
```diff
@@ -47,6 +47,7 @@ export default function SnackbarProvider({
   children,
   overrides = {},
   placement,
+  defaultDuration,
```
Comment (chasestarr) [inline src/snackbar/snackbar-context.js:50]:
```suggestion
  defaultDuration = DURATION.short,
```

File: src/snackbar/snackbar-context.js [hunk 2]
```diff
@@ -58,7 +59,7 @@ export default function SnackbarProvider({
   const [containerHeight, setContainerHeight] = React.useState(0);
   const containerRef = React.useRef(null);
 
-  function enqueue(elementProps, duration = DURATION.short) {
+  function enqueue(elementProps, duration = defaultDuration || DURATION.short) {
```
Comment (chasestarr) [inline src/snackbar/snackbar-context.js:62]:
```suggestion
  function enqueue(elementProps, duration = defaultDuration) {
```

File: src/snackbar/snackbar-context.js (hunk 1 above)
Comment (chasestarr) [reply src/snackbar/snackbar-context.js:50]: This way it only applies short if `defaultDuration` is undefined

**Review by chasestarr — COMMENTED** [review]
Thanks for making this change!

**Review by chasestarr — APPROVED** [review]
Thank you

**Pull request merged**